cpp/velox/substrait/SubstraitToVeloxPlanValidator.cc

```diff
@@ -263,8 +263,16 @@ bool SubstraitToVeloxPlanValidator::isAllowedCast(const TypePtr& fromType, const
 
   // Limited support for X to Timestamp.
   if (toType->isTimestamp() && !(fromType->isDate() || fromType->isVarchar())) {
-    LOG_VALIDATION_MSG("Casting from " + fromType->toString() + " to TIMESTAMP is not supported.");
-    return false;
+    switch (fromType->kind()) {
+      case TypeKind::TINYINT:
```

Comment:
can we move `fromType->isDate() || fromType->isVarchar()` to here?

Reply:
@jackylee-ch, just moved and also use isXXX() to replace TypeKind check.